lighthouse-core/lib/network-request.js

```diff
@@ -297,12 +297,24 @@ module.exports = class NetworkRequest {
 
   /**
    * LR loses transfer size information, but passes it in the 'X-TotalFetchedSize' header.
+   * 'X-TotalFetchedSize' is the canonical transfer size in LR. Nothing should supersede it.
+   *
+   * The total length of the encoded data is spread out among multiple events. The sum of the
+   * values in onResponseReceived and all the onDataReceived events typically equals the value
+   * seen on the onLoadingFinished event. In <1% of cases we see the values differ. As we process
+   * onResonseReceived and onDataReceived we accumulate the total encodedDataLength. When we
+   * process onLoadingFinished, we override the accumulated total. We do this so that if the
+   * request is aborted or fails, we still get a value via the accumulation.
```

Comment:
> or fails

did we decide not to override for `onLoadingFailed`?

Reply:
`_updateTransferSizeForLightrider` is being called on failed now, or did you mean  something else?

Reply:
oh I just missed it and comment didn't mention it so I got confused, we're all good :)